### PR TITLE
Implement liking

### PR DESCRIPTION
**Pull request by cedricongjh**
Implement liking logic and fetch user likes in profile page

### CODE REVIEW

**Review by MarcusTXK — COMMENTED** [review]
Not sure if its just me, but I cant get it running on my side to test

image

Looked through the logic itself and looks good. I didn't know that we could get their profile info that way, so there is not need to store their email. Thanks for helping to fix that.

Lastly, could you remove .flutter-plugins from the repo?

**Comment by cedricongjh** [comment]
@MarcusTXK I updated the PR, forgot to commit one file, can try to see if it runs now
.flutter-plugins has also been removed (it's already in .gitignore)

**Comment by MarcusTXK** [comment]
It doesnt seem to fix it:
image

Here is one of the lines:
image

**Comment by cedricongjh** [comment]
@MarcusTXK ok can try once again, sorry missed out another file to commit

**Review by MarcusTXK — APPROVED** [review]
LGTM, Tested and works great, thanks alot for your help

File: lib/Pages/ViewItem.dart [hunk 1]
```diff
@@ -74,8 +75,7 @@ class _ViewItemState extends State<ViewItem> {
       date = item.date;
       owner = item.owner;
 
-      // TODO fetch from user database
-      email = user!.email;
+      email = provider.user?.email ?? "";
```
Comment (MarcusTXK) [inline lib/Pages/ViewItem.dart:78]: Wait this should be the email of the owner of the item, not the current user's email

Comment (cedricongjh) [reply lib/Pages/ViewItem.dart:78]: ah ok, i just pushed a commit to add email to userstate, copying the email from the auth to the document

**Pull request merged**